File: onlinejudge/leetcode/455-Assign Cookies.go
func findContentChildren(g []int, s []int) int {
	sort.Ints(g)
	sort.Ints(s)

	childI := 0
	cookieJ := 0

	for childI < len(g) && cookieJ < len(s) {
		if s[cookieJ] >= g[childI] {
			childI += 1
		}
		cookieJ += 1
	}

	return childI
}

File: onlinejudge/leetcode/455-Assign Cookies.java
class Solution {
    public int findContentChildren(int[] g, int[] s) {
        Arrays.sort(g);
        Arrays.sort(s);

        int childI = 0; 
        int cookieJ = 0;

        while (childI < g.length && cookieJ < s.length) {
            if (s[cookieJ] >= g[childI]) {
                childI += 1;
            }
            cookieJ += 1;
        }
        return childI;
    }
}
